javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -319,7 +323,7 @@ private Vector(Class<?> type, Object leading, VectorTree<T> middle, Object trail
      */
     public static Vector<Integer> ofAll(int[] array) {
         Objects.requireNonNull(array, "array is null");
-        return ofAll(Iterator.ofAll(array));
+        return getTs(array, array.getClass().getComponentType(), array.length);
```

Comment:
Here the values are not boxed at all (if the provided specialized methods are used for retrieval)